loopy/transform/merge_arrays.py

```diff
@@ -0,0 +1,113 @@
+__copyright__ = "Copyright (C) 2022 Isuru Fernando"
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+__doc__ = """
+.. currentmodule:: loopy
+.. autofunction:: merge_temporary_arrays
+"""
+
+from loopy.kernel.data import auto
+from loopy.symbolic import SubstitutionRuleMappingContext
+from loopy.kernel import LoopKernel
+from loopy.translation_unit import for_each_kernel
+
+import pymbolic.primitives as prim
+from pytools import all_equal
+
+from typing import Text, Sequence, Optional, Union
+
+
+@for_each_kernel
+def merge_temporary_arrays(
+        kernel: LoopKernel,
+        array_names: Union[Text, Sequence[Text]],
+        new_name: Optional[Text] = None,
+        axis_nr: int = 0) -> LoopKernel:
+    """Merges temporary arrays into one array along the axis given by *axis_nr*.
+
+    :arg array_names: a list of names of temporary variables or arguments. May
+        also be a comma-separated string of these.
+
+    :arg axis_nr: the (zero-based) index of the axis of the arrays to be merged.
+
+    :arg new_name: new name for the merged temporary. If not given, the name
```

Comment:
Maybe generate a new name, instead of picking `array_names[0]`?

Reply:
Done in da48e89